.claude/hooks/check_line_count_hook.py

```
@@ -20,21 +21,38 @@ def main():
+        with open(file_path, encoding="utf-8") as f:
             line_count = sum(1 for line in f)
-
         if line_count > MAX_LINES:
-            print(f"Suggestion: File '{file_path}' has {line_count} lines, which exceeds the recommended {MAX_LINES} lines. Consider refactoring this file to improve readability and maintainability.", file=sys.stderr)
-
-    except Exception as e:
+            print(
+                f"Suggestion: File '{file_path}' has {line_count} lines, which exceeds the recommended {MAX_LINES} lines. Consider refactoring this file to improve readability and maintainability.",
+                file=sys.stderr,
+            )
+    except (IOError, OSError, UnicodeDecodeError) as e:
```

Ruff (0.11.9)

Replace aliased errors with `OSError`

Replace with builtin `OSError`

(UP024)